price-calc/src/main/java/by/prohorov/price/kg/CostByPriceKgUseFileXml.java

```diff
@@ -0,0 +1,47 @@
+package by.prohorov.price.kg;
+
+import by.prohorov.price.entity.Price;
+import by.prohorov.price.parser.Jackson;
+import by.prohorov.validate.Validator;
+
+import java.io.IOException;
+
+/**
+ * Created by Artsiom Prokharau 09.02.2021
+ */
+
+public class CostByPriceKgUseFileXml implements CostByPriceKg {
+
+    private double costWeight;
+
+    public CostByPriceKgUseFileXml() {
+    }
+
+    public CostByPriceKgUseFileXml(Validator validator) {
+        this.costWeight = validator.checkValue("Please, enter weight (kg) : =>  ");
+    }
+
+    public double getCostWeight() {
+        return costWeight;
+    }
+
+    public void setCostWeight(double costWeight) {
+        this.costWeight = costWeight;
+    }
+
+    @Override
+    public double costByPriceWeightInKg() {
+        try {
+            if (costWeight < 4) {
+                costWeight = Jackson.loaderXmlFile().getLight();
```

Comment:
how will you handle the case when `Jackson` will not be able to load the needed data

Reply:
Add check Exception to class Jackson

Reply:
I don't like a lot of try-catch/exception, why I don't want for example set integers = 0 (by default), then you can return an empty `new Price()`. I mean you will not check nulls and will avoid unnecessary try-catch